src/ToolBox/superpmi/mcs/verbmerge.cpp

```diff
@@ -15,42 +15,52 @@
 // The caller must delete the returned string with delete[].
 //
 // static
-char* verbMerge::MergePathStrings(const char* dir, const char* file)
+LPTSTR verbMerge::MergePathStrings(LPCTSTR dir, LPCTSTR file)
 {
-    size_t dirlen  = strlen(dir);
-    size_t filelen = strlen(file);
+    size_t dirlen  = wcslen(dir);
+    size_t filelen = wcslen(file);
     size_t newlen  = dirlen + 1 /* slash */ + filelen + 1 /* null */;
-    char*  newpath = new char[newlen];
-    strcpy(newpath, dir);
-    strcat(newpath, DIRECTORY_SEPARATOR_STR_A);
-    strcat(newpath, file);
+    LPTSTR newpath = new WCHAR[newlen];
+    wcscpy(newpath, dir);
+    wcscat(newpath, DIRECTORY_SEPARATOR_STR_W);
+    wcscat(newpath, file);
     return newpath;
 }
 
+char* verbMerge::ConvertUTF8ToMultiByte(LPCTSTR wstr)
+{
+    unsigned int codePage   = CP_UTF8;
+    int          sizeNeeded = WideCharToMultiByte(codePage, 0, wstr, -1, NULL, 0, NULL, NULL);
+    char*        encodedStr = new char[sizeNeeded];
+    WideCharToMultiByte(codePage, 0, wstr, -1, encodedStr, sizeNeeded, NULL, NULL);
+    return encodedStr;
+}
+
```

Comment:
Looks like this is only used for Log output? In that case, can't you just use `%S` (capital 'S') instead of `%s` (lower-case 's') in the format specification, instead?